### DISCUSSION TITLE
Access to Insiders for multiple people at org?

### DISCUSSION THREAD

**Discussion by StarfallProjects** [squidfunk/mkdocs-material#3816] [Q&A] [answered]
Hi! We're sponsoring as an organization and have created a bot account as instructed. It seems to be working - I can use its token locally and on Netlify. However, a colleague is now trying to get set up to run it locally and is getting errors at pip install:

```
WARNING: Discarding git+https://***@github.com/squidfunk/mkdocs-material-insiders.git. Command errored out with exit status 128: git clone -q 'https://***@github.com/squidfunk/mkdocs-material-insiders.git' /tmp/pip-req-build-owufnupz Check the logs for full command output.
ERROR: Command errored out with exit status 128: git clone -q 'https://****@github.com/squidfunk/mkdocs-material-insiders.git' /tmp/pip-req-build-owufnupz Check the logs for full command output.
```

We were wondering if you also had to be signed in with the linked github account - but Netlify isn't (though it could be getting away with it because it's signed in with my account, which also sponsors . . . )

**Comment by squidfunk** [comment, marked as answer]
Thanks for asking. The general advice for organizations is to fork the Insiders repository to your organization, which allows you to grant access to all members of your organization. It's currently not possible for me to add multiple accounts, but I'm working on a solution so organizations can add more collaborators by themselves without the necessity for forking. [1]

In the meantime, forking is the way to go. GitHub makes it easy to [stay in sync with upstream](https://stackoverflow.com/a/67425996/1065584). __Thus, you can currently only pull changes with the account that was added as a collaborator, so make sure that you're signed in with that account.__

[1]: GitHub recently announced (and [released](https://github.blog/2022-02-02-new-sponsors-only-repositories-custom-amounts-and-more/)) __sponsor-only repositories__, which would theoretically make my custom webhook-based integration obsolete, so I was really looking forward on their take on this, but sadly it only works for individual collaborators. Same thing as my custom integration. I'm kind of hoping that GitHub comes up with a solution for this, but I guess I have to build something myself again.

**Reply by StarfallProjects** [reply]
Ok so just to check:

1. Log in as collaborator
2. Fork Insiders to the org
3. Everyone can now install it by the git clone & install method

**Reply by squidfunk** [reply]
Jup, correct. Remember to create a _private_ fork, but I think GitHub won't allow you to create a public fork if the repository is private anyway. I think forking should be enough to grant access to everybody inside your organization (unless you have further restricted access inside your org, e.g. by forbidding specific users to view specific repositories).

**Comment by s10doc** [comment]
Hello, 

I am trying to install 'Insiders' and getting errors with latest generated token.  I am a sponsor, paying monthly. 

C:\Users\jacquelinel>pip install git+https://$ghp_REDACTED@github.com/squidfunk/mkdocs-material-insiders.git
Collecting git+https://****@github.com/squidfunk/mkdocs-material-insiders.git
  Cloning https://****@github.com/squidfunk/mkdocs-material-insiders.git to c:\users\jacquelinel\appdata\local\temp\pip-req-build-at0ohzxu
  Running command git clone --filter=blob:none --quiet 'https://****@github.com/squidfunk/mkdocs-material-insiders.git' 'C:\Users\jacquelinel\AppData\Local\Temp\pip-req-build-at0ohzxu'
  bash: /dev/tty: No such device or address
  error: failed to execute prompt script (exit code 1)
  fatal: could not read Password for 'https://$ghp_REDACTED@github.com': No such file or directory
  error: subprocess-exited-with-error

  × git clone --filter=blob:none --quiet 'https://****@github.com/squidfunk/mkdocs-material-insiders.git' 'C:\Users\jacquelinel\AppData\Local\Temp\pip-req-build-at0ohzxu' did not run successfully.
  │ exit code: 128
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× git clone --filter=blob:none --quiet 'https://****@github.com/squidfunk/mkdocs-material-insiders.git' 'C:\Users\jacquelinel\AppData\Local\Temp\pip-req-build-at0ohzxu' did not run successfully.
│ exit code: 128
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.


Can you help me, please. 

Kind regards, 
Jacqueline

__Edit__ by @squidfunk: redacted tokens.

**Reply by squidfunk** [reply]
__So first of all, you should never share your `GH_TOKEN` publicly, which you did in your post, so I strongly recommend regenerating the token. Otherwise other users now have access to your entire account__.

My guess is that the `$` is causing the problem:

```
git+https://$ghp_...@github.com/squidfunk/mkdocs-material-insiders.git
```

if you remove that, it should work, given that you have given the correct scopes to the token.

**Reply by s10doc** [reply]
Thank you very much for protecting us... my bad for exposing the full token.   

Really appreciate your prompt action!  Thank you so much!

**Reply by squidfunk** [reply]
No worries, here to help! Regarding the problem you reported – did removing the `$` help? Another clue that this is the error is the URL that Python is trying to clone, which is missing the token:

```
Cloning https://@github.com/squidfunk/mkdocs-material-insiders.git
                ^ missing token
```

**Reply by s10doc** [reply]
Thank you, I have instead opted for downloading the zip file and then installing ...   we will update once/month  so no need for the token

**Comment by HonkingGoose** [comment]
> Edit by `@squidfunk`: redacted tokens.

You redacted the tokens in the _last_ comment, but the comment edit history still has them. 😉

Read these pages to learn how to redact the tokens fully:

- [GitHub docs, redacting sensitive information](https://docs.github.com/en/communities/moderating-comments-and-conversations/managing-disruptive-comments#redacting-sensitive-information)
- [GitHub docs, deleting sensitive information from a comment's history](https://docs.github.com/en/communities/moderating-comments-and-conversations/tracking-changes-in-a-comment#deleting-sensitive-information-from-a-comments-history)

Just letting you know. I assume that the original poster has regenerated their token by now.

**Reply by squidfunk** [reply]
Thanks! I didn't know that this is possible. Always great to learn something new. I've deleted the revisions.
